### DISCUSSION TITLE
Static and Dynamic Friction

### DISCUSSION THREAD

**Discussion by GloriousPtr** [jrouwe/JoltPhysics#237] [Q&A] [answered]
Nvidia PhysX provides support for static and dynamic friction.
Whereas as far as I can tell Jolt only has one friction parameter per body presumably dynamic friction (or the same value is used for both static and dynamic friction) Can this be replicated in Jolt Physics?

Nvidia PhysX implementation:
```cpp
PxMaterial* mMaterial;
mMaterial = mPhysics->createMaterial(0.5f, 0.5f, 0.1f); // static friction, dynamic friction,
                                                        // restitution
```

**Comment by jrouwe** [comment, marked as answer]
The difference between static and dynamic friction is currently not supported by Jolt. I considered implementing it (and I may make it an official feature in the future), but I think the difference between these two friction types is quite subtle. Yes, you will be able to see it if you put a cube on a flat plane, but in a more complex environment I think it is hard to notice. 

If you need it, you can emulate it by overriding the OnContactAdded/Persisted methods and override ioSettings.mCombinedFriction. You'd calculate the velocity difference between the bodies (preferably at the contacts of the manifold using e.g. Body::GetPointVelocity) to see if the bodies are moving relative to each other or not. If they're not moving (with a certain threshold) you return the combined static friction, if they are moving you return the combined dynamic friction.

**Reply by GloriousPtr** [reply]
I agree that it's not quite noticeable but I'm making an engine and I think it's a good to have.
Thanks for the neat explanation about implementation details :)
